pkg/kapp/logs/pod_log.go

```diff
@@ -41,12 +42,24 @@ func (l PodLog) TailAll(ui ui.UI, cancelCh chan struct{}) error {
 
 	for _, cont := range l.pod.Spec.InitContainers {
 		if !(podInTerminalState && l.isWaitingContainer(cont, l.pod.Status.InitContainerStatuses)) {
+			if len(l.opts.ContainerNames) > 0 {
+				if l.isContainerInSlice(cont, l.opts.ContainerNames) {
+					conts = append(conts, cont)
+				}
+				continue
+			}
 			conts = append(conts, cont)
 		}
 	}
 
 	for _, cont := range l.pod.Spec.Containers {
 		if !(podInTerminalState && l.isWaitingContainer(cont, l.pod.Status.ContainerStatuses)) {
+			if len(l.opts.ContainerNames) > 0 {
```

Comment:
given change below in isContainerInSlice, we can simplify to append if isContainerInSlice return true.